src/auth/index.js

```diff
@@ -7,6 +7,27 @@ import * as common from '../common';
 
 const jwtDecode = require('jwt-decode');
 
+const DEFAULT_ACCESS_TOKEN_EXPIRE_WITHIN_SECS = 10;
+
+const ACCESS_TOKEN_FIELD = 'accessToken';
+const REFRESH_TOKEN_FIELD = 'refreshToken';
+const DEVICE_ID_FIELD = 'deviceId';
+const USER_ID_FIELD = 'userId';
+
+const clientJSONFields = {
+  [ACCESS_TOKEN_FIELD]: 'accessToken',
+  [REFRESH_TOKEN_FIELD]: 'refreshToken',
+  [DEVICE_ID_FIELD]: 'deviceId',
+  [USER_ID_FIELD]: 'userId'
+};
+
+const adminJSONFields = {
+  [ACCESS_TOKEN_FIELD]: 'access_token',
+  [REFRESH_TOKEN_FIELD]: 'refresh_token',
+  [DEVICE_ID_FIELD]: 'device_id',
+  [USER_ID_FIELD]: 'user_id'
```

Comment:
I don't think these should be in `auth` I think these should be passed in when `auth` is constructed and then that would set `jsonFields` on `auth`.

Reply:
The main issue is really that `Auth` has a `this.client` and `Client` has a `this.auth`

Reply:
I disagree that these should be passed in. Auth itself has ownership of the logic here. It would be poor encapsulation to pass that configuration in.

Reply:
Auth receiving a client is unrelated to this